homeassistant/components/homewizard_energy/coordinator.py

```diff
@@ -0,0 +1,93 @@
+"""Update coordinator for HomeWizard Energy."""
+
+from datetime import timedelta
+import logging
+
+import aiohwenergy
+import async_timeout
+
+from homeassistant.const import CONF_API_VERSION, CONF_ID, CONF_NAME, CONF_STATE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    ATTR_BRIGHTNESS,
+    ATTR_POWER_ON,
+    ATTR_SWITCHLOCK,
+    CONF_DATA,
+    CONF_MODEL,
+    CONF_SW_VERSION,
+    MODEL_P1,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class HWEnergyDeviceUpdateCoordinator(DataUpdateCoordinator):
+    """Gather data for the energy device."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        api: aiohwenergy.HomeWizardEnergy,
+    ) -> None:
+        """Initialize Update Coordinator."""
+
+        self.api = api
+
+        update_interval = self.get_update_interval()
+        super().__init__(hass, _LOGGER, name="", update_interval=update_interval)
+
+    def get_update_interval(self) -> timedelta:
+        """Return best interval for product type."""
+        try:
+            product_type = self.api.device.product_type
+        except AttributeError:
+            product_type = "Unknown"
+
+        if product_type == MODEL_P1:
+            try:
+                smr_version = self.api.data.smr_version
+                if smr_version == 50:
+                    return timedelta(seconds=1)
```

Comment:
We do not allow polling API update intervals lower than 5 seconds.

Reply:
Fixed by [14897528a](https://github.com/DCSBL/core/commit/14897528ab2c4441b69f21177c505372a2fc2796) and [9cf9c0e7a](https://github.com/DCSBL/core/commit/9cf9c0e7a26000af2cd4d132c98a1ee81044374a)